cmp/filter/FilterChooserFieldSpec.js

```diff
@@ -51,7 +50,66 @@ export class FilterChooserFieldSpec {
     exampleValue;
 
     /** @return {FieldType} */
-    get fieldType() {return this.field.type}
+    fieldType;
+
+    /**
+     * @param {Object} c - FilterChooserFieldSpec configuration.
+     * @param {String} c.field - name of field configured by this class. If the controlling
+     *      FilterChooserModel has been configured with a `sourceStore` having a `Field` with the
+     *      same name, this class will use that `Field` to default several of the configs below.
+     * @param {Object} [c.fieldType] - type of field, defaulted from matching Store field if
+     *      available, or 'AUTO'.
+     * @param {string} [c.displayName] - user-facing / longer name for display, defaulted from
+     *      matching Store field if available.
+     * @param {string[]} [ops] - operators available for filtering, defaulted to a supported set
+     *      based on `fieldType`.
+     * @param {boolean} [suggestValues] - true to provide auto-complete options with data values
+     *      sourced either automatically from Store data or provided directly via the `values`
+     *      config below. Default is based on fieldType. Set to `false` to disable extraction and
+     *      suggestion of values from Store.
+     * @param {[]} [values] - explicit list of available values to autocomplete for this field.
+     *      If unspecified, values will be extracted and updated from Store data if applicable.
+     * @param {FilterOptionValueRendererCb} [valueRenderer] - function to produce a suitably
+     *      formatted string for display to the user for any given field value.
+     * @param {FilterOptionValueParserCb} [valueParser] - function to parse user's input from a
+     *      filter chooser control into a typed data value for use in filtering comparisons.
+     * @param {*} [exampleValue] - sample / representative value displayed by `FilterChooser`
+     *      components to aid usability.
+     * @param {Store} [store] - set from controlling `FilterChooserModel.sourceStore` config, used
+     *      to source matching data `Field` and extract values if configured.
+     */
```

Comment:
I had specifically documented these via a typedef so we could then use it in the documentation for `FilterChooserModel.fieldSpecs` - the actual entry point for the developer. I had done this for `FieldConfig` as well (in Field.js).

IMO pushing it back to the ctor here just encourages a dev to construct this object, which we don't want them to do, and doesn't provide them any inline help when specifying fieldSpecs, which is exactly what we _do_ need them to do.

I left this in its new location on the ctor and updated the tags to `@param` consistently but would like to discuss as this is a pattern we have in a few places where we have really nice docs, but they aren't easily discoverable and don't provide the dev-time assistance they could because they are on a ctor that devs never call (`Column` is another prime example).